### PR TITLE
Fix double network failure handling

### PR DESCRIPTION
**Pull request by surabujin**
Add queue in front of reroute "handler" so if reroute for the flow is
going and new reroute request for this flow received (double network
failure case) second request will be postponed.

At the end of reroute request processing "handler" check presence of
postponed request and handle them.

---
This change is [Reviewable](https://reviewable.io/reviews/telstra/open-kilda/3108)

### CODE REVIEW

**Comment by surabujin** [comment]
Related to #3087

File: services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java [hunk 1]
```diff
@@ -67,33 +72,51 @@ public RerouteService(PersistenceManager persistenceManager) {
...
     public void rerouteAffectedFlows(MessageSender sender, String correlationId, RerouteAffectedFlows command) {
+        // TODO(surabujin): need better/more detailed representation of failed ISL
         PathNode pathNode = command.getPathNode();
         int port = pathNode.getPortNo();
         SwitchId switchId = pathNode.getSwitchId();
+        final IslEndpoint affectedIsl = new IslEndpoint(switchId, port);
+
         Collection<FlowPath> affectedFlowPaths
                 = getAffectedFlowPaths(pathNode.getSwitchId(), pathNode.getPortNo());
+        Map<String, Flow> flowById = collectAffectedFlows(affectedFlowPaths);
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java]: what's the purpose of such aggregation?

File: services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java [hunk 2]
```diff
@@ -67,33 +72,51 @@ public RerouteService(PersistenceManager persistenceManager) {
...
-        Map<Flow, Set<PathId>> flowsForRerouting = groupFlowsForRerouting(affectedFlowPaths);
-        for (Entry<Flow, Set<PathId>> entry : flowsForRerouting.entrySet()) {
-            Flow flow = entry.getKey();
+
+        Map<String, List<FlowPath>> flowsForRerouting = groupPathsForRerouting(affectedFlowPaths);
+        for (Entry<String, List<FlowPath>> entry : flowsForRerouting.entrySet()) {
+            String flowId = entry.getKey();
+            List<FlowPath> flowPaths = entry.getValue();
+
+            Flow flow = flowById.get(flowId);
```
Comment (timofei-durakov) [inline services/wfm/src/main/java/org/openkilda/wfm/topology/reroute/service/RerouteService.java]: could you please comment out a scenario for this

File: services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/FlowRerouteRequest.java [hunk 3]
```diff
@@ -41,29 +41,37 @@
...
+    protected Set<IslEndpoint> affectedIsl;
 
     /**
      * Update flow even if path will not be changed.
      */
     @JsonProperty("force")
     private boolean force;
 
+    @JsonProperty("effectively_down")
+    private boolean effectivelyDown;
```
Comment (timofei-durakov) [inline services/src/messaging/src/main/java/org/openkilda/messaging/command/flow/FlowRerouteRequest.java]: what does it mean? leave a doc

**Pull request merged**